content/docs/user-guide/programming/gems/code-gems.md

```diff
@@ -43,3 +43,21 @@ The `<Gem>EditorModule` class contains the logic for using the Gem in the Editor
 ### `<Gem>EditorSystemComponent.cpp`, `<Gem>EditorSystemComponent.h`
 
 The `<Gem>EditorSystemComponent` class is the system component class for managing the Editor module. You can use this class to program Editor functionality. For example, if your Gem implements a component or a window that appears in the O3DE Editor.
+
+## Building a Gem
+
+After a Gem is registered to an engine, project, or the global `o3de_manifest.json`, you can build the gem using the same command line tool that you use to [build the engine and projects](/docs/user-guide/build/).  Gems that have been added to a project will be built automatically when the project builds.  While developing a Gem, however, you may want to build it individually to avoid rebuilding the entire project.
+
+In the directory that is appropriate for your [engine installation](/docs/welcome-guide/setup/setup-from-github/building-windows/#build-the-engine), use the `cmake --build` command and specify the build solution and configuration.  Use the `--target` flag to specify the Gem modules that you want to build, `<Gem Name>` for the main Gem module or `<Gem Name>.Editor` for Editor modules.  
+
+The following example shows the command to build _both_ the Editor and main Gem modules for a Gem named `MyGem`:
+
+```cmd
+cmake --build build/windows --target MyGem.Editor MyGem --config profile -- -m
+```
+
+When you build a Gem, the build system will also build any of the Gem's dependencies that are listed in the Gem's `code/CMakeLists.txt` file.
+
+### Building a Gem in Visual Studio
+
+Open the build solution that is appropriate your [engine installation](/docs/welcome-guide/setup/setup-from-github/building-windows/#build-the-engine), for source engine installations this will be the engine's root directory, for pre-built SDK engines this will be the project's root directory. Then find your Gem's `Code` directory in the Solution Explorer.  This directory contains all of your Gem's build targets, including the main `<MyGem>` and Editor `<MyGem>.Editor` modules.  Ensure that the correct build configuration is selected on the top toolbar. **Right-click** on a module and select **Build** from the menu to build the module.  
```

Comment:
It looks like there is a missing `for` before `your`.

Also for the second sentence which starts with "Then find your Gem's `Code` directory in the Solution Explorer", we may want to clarify it that it is at `<MyGem>/Code`
![image](https://user-images.githubusercontent.com/56135373/200684783-2d840ad6-5547-4a41-9e01-42e540812dc0.png)

Reply:
I fixed the type and added an image of the context menu after right-clicking a build target.  I also added a clarification that the code directory would _in most cases_ be found at `Gems\<MyGem>\Code`.  I noticed that some of the core Gems have an additional layer of directories before you hit `Code` (an example is `Atom\Asset\ImageProcessingAtom\Code`) and I'm guessing that it's possible for a user to create a Gem in a similar manner.  Is that correct?

Reply:
Yes, the Atom gem contains other gems within them.
So Atom contains an `ImageProcessingAtom` gem within it located at Atom/Asset/ImageProcessingAtom

We call these sub gems and they are listed in the `gem.json` files for the parent gems in the  "[external_subdirectories](https://github.com/o3de/o3de/blob/development/Gems/Atom/gem.json#L19-L31)" field.